### ISSUE TITLE
Check AMD GPU support

### ISSUE THREAD

**Issue by jatkinson1000** [Cambridge-ICCS/FTorch#128]
@TomMelt suggested we check support on AMD GPU devices and offered to test on a suitable machine.

To do this may require modifications akin to those in #125.

**Comment by jatkinson1000** [comment]
I think this would require downloading ROCM compatible rather than CUDA compatible libtorch as described here: https://pytorch.org/blog/pytorch-for-amd-rocm-platform-now-available-as-python-package/

We would then need to add the relevant enum to target the AMD backend.

This is perhaps something that could be combined in #125, or at least follow a very similar pattern.

**Comment by jatkinson1000** [comment]
@jwallwork23 I don't think we can escape `CUDA` as the LibTorch approach, but having an FTorch `AMD` or `ROCM` device that maps to the CUDA LibTorch enum would be nice I think.

**Comment by TomMelt** [comment]
@jwallwork23 @jatkinson1000  Yeah my intention for now is to submit a PR that maps `HIP` (matching the PyTorch enum naming scheme) to `CUDA` so that if a user specifies `HIP` it will actually use the `CUDA` backend. We will document this to inform users. Potentially in the future, if AMD moves away from the `CUDA` backend to their own `HIP` backend we can revive PR #384

**Comment by jwallwork23** [comment]
WIP on branch https://github.com/Cambridge-ICCS/FTorch/tree/128_amd-gpu.

**Comment by jatkinson1000** [comment]
Reopening and linking #387
